src/renderer/api/kube-object.ts

```diff
@@ -40,6 +40,26 @@ export interface IKubeObjectMetadata {
   }[];
 }
 
+export interface IKubeStatus {
+  kind: string;
+  apiVersion: string;
+  code: number;
+  message?: string;
+  reason?: string;
+}
+
+export class KubeStatus implements IKubeStatus {
+  public readonly kind: string;
+  public readonly apiVersion: string;
+  public readonly code: number;
+  public readonly message: string;
+  public readonly reason: string;
+
+  constructor(data: IKubeStatus) {
+    Object.assign(this, data);
+  }
+}
+
```

Comment:
What is the benefit of this class vs the interface above? also this contructor is incorrect as `message` and `reason` should either be optional or be defaulted to `""`

Reply:
Class because then user can check `xyz instanceof KubeStatus` ... no other reason really.

Reply:
Okay sounds good, though I still think that we shouldn't use `Object.assign` in constructors.

Reply:
What should we use instead?

Reply:
Assigning directly,

```typescript
export class KubeStatus implements IKubeStatus {
  public readonly kind: string;
  public readonly apiVersion: string;
  public readonly code: number;
  public readonly message: string;
  public readonly reason: string;

  constructor(data: IKubeStatus) {
    this.kind = data.kind;
    this.apiVersion = data.apiVersion;
    this.code = data.code;
    this.message = data.message || "";
    this.reason = data.reason || "";
  }
}
```

We only have to write this code once, so I don't see why we should be copying over more than what we actually want.

Reply:
Agreed and fixed.